main.go

```diff
@@ -14,191 +16,328 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+const concurrentFetch = 100
+
+// Commandline flags.
 var (
-	configFile  = flag.String("config.file", "", "Path to config file.")
-	metricsPath = flag.String("web.telemetry-path", "/metrics", "Path under which to expose metrics.")
+	addr           = flag.String("web.listen-address", ":9105", "Address to listen on for web interface and telemetry")
+	autoDiscover   = flag.Bool("exporter.discovery", false, "Discover all Mesos slaves")
+	localAddr      = flag.String("exporter.local-address", "127.0.0.1:5051", "Address to connect to the local Mesos slave")
+	masterURL      = flag.String("exporter.discovery.master", "http://mesos-master.example.com:5050", "Mesos master URL")
+	metricsPath    = flag.String("web.telemetry-path", "/metrics", "Path under which to expose metrics")
+	scrapeInterval = flag.Duration("exporter.interval", (60 * time.Second), "Scrape interval duration")
 )
 
 var (
-	cpuLimitVal = prometheus.NewGaugeVec(
-		prometheus.GaugeOpts{
-			Namespace: "mesos_task",
-			Name:      "cpu_limit",
-			Help:      "Fractional CPU limit.",
-		},
-		[]string{"task", "mesos_slave", "framework_id"},
+	variableLabels = []string{"task", "mesos_slave", "framework_id"}
```

Comment:
To be consistent with the other "slave" change now, let's also remove the `mesos_` prefix in these labels? Just `slave` and `task`, I'd say...